### PR TITLE
extra configuration options

### PR DESCRIPTION
**Pull request by mihailogazda**
Adding:

- `applicationId` - adding app ID for cpp apps
- `androidprojectdependencies`- add dependency to internal project - project scope 
- `inject2cmake` - extra cpp compile arguments for cmake
- Google Service JSON support

### CODE REVIEW

File: android_studio/android_studio.lua [hunk 1]
```diff
@@ -194,6 +194,22 @@ function m.generate_manifest(prj)
...
     p.pop('</manifest>')
 end
+
+function m.generate_googleservices(prj)
+    -- look for a google-services in project files
+    for cfg in project.eachconfig(prj) do
+        for _, file in ipairs(cfg.files) do
+            if string.find(file, "google.services.json") then
+                -- copy contents of google-services and write with premake
+                jsonFile = io.open(file, "r")
```
Comment (polymonster) [inline android_studio/android_studio.lua:204]: use snake_case for these variable names

File: android_studio/_preload.lua [hunk 2]
```diff
@@ -112,6 +113,13 @@ p.api.register
     kind = "list:string"
 }
 
+p.api.register 
+{
+	name = "androidprojectdependencies",
```
Comment (polymonster) [inline android_studio/_preload.lua:118]: indent with 4 spaces, as tabs appear larger in browser

**Review by polymonster — CHANGES_REQUESTED** [review]
Thanks for this, just some small changes requested for consistency.

snake_case or all lowercase naming for variables in the source code, 4 spaces instead of tabs to indent.

for the new setting names, I think they should be consistent with the others, and while a bit of a mouthful with all lower case (as the rest of premake) maybe something along these lines:

`applicationId` to `androidappid`?
`inject2cmake` to `androidcmake`?

something along those lines to scope them to android, as there might be other plugins which also use cmake or appids.

Hope this is OK! and thanks for contributing... we also use google-services for firebase but have another build step which makes the copy, so I am looking forward to trying this out and doing it all via premake.

**Comment by mihailogazda** [comment]
Cheers mate, can you take a look now?

**Review by polymonster — APPROVED** [review]
looks good, thanks :)

**Pull request merged**